generate/convert.go

```diff
@@ -264,19 +265,52 @@ func (g *generator) convertDefinition(
 			GoName:          name,
 			Description:     def.Description,
 			GraphQLName:     def.Name,
+			SharedFields:    make([]*goStructField, 0, len(selectionSet)),
 			Implementations: make([]*goStructType, len(implementationTypes)),
 		}
 		g.typeMap[name] = goType
 
+		// TODO(benkraft): This sorta-duplicates what we'll do in each
+		// implementation when it traverses the fields.  But they'll differ
+		// more once we support fragments; at that point we should figure out
+		// how to refactor.
+		for _, selection := range selectionSet {
+			field, ok := selection.(*ast.Field)
+			if !ok { // fragment/interface, not a shared field
```

Comment:
As we discovered with the validator edge case, a union doesn't have shared fields, but it may be possible to treat the fields as shared if all the union members implement the same interface corresponding to a fragment. Anyway, this will be tackled in the upcoming fragment changes (if we want to tackle it at all).